llvm/lib/ExecutionEngine/JITLink/ELFLinkGraphBuilder.h

```diff
@@ -397,6 +397,11 @@ template <typename ELFT> Error ELFLinkGraphBuilder<ELFT>::graphifySections() {
                                   orc::ExecutorAddr(Sec.sh_addr),
                                   Sec.sh_addralign, 0);
 
+    if (Sec.sh_type == ELF::SHT_ARM_EXIDX) {
+      // Add live symbol to avoid dead-stripping for .ARM.exidx sections
+      G->addAnonymousSymbol(*B, orc::ExecutorAddrDiff(), orc::ExecutorAddrDiff(), false, true);
```

Comment:
The compiler emits a `R_ARM_NONE` relocation from an exception-handling table section to the required personality routine to indicate the dependence, but the edge alone doesn't keep the section alive in JITLink. We add the anonymous live symbol here to preserve it from dead-stripping.

Reply:
That's surprising -- any valid edge should keep the routine alive? Do we actually build an edge, or do we discard `R_ARM_NONE` relocations?

Reply:
IIUC the issue is that here https://github.com/llvm/llvm-project/blob/release/17.x/llvm/lib/ExecutionEngine/JITLink/JITLinkGeneric.cpp#L271 we start from a set of live symbols and then follow edges and mark their target blocks live. Blocks without any symbols and only outgoing edges will remain dead.

I implemented `R_ARM_NONE` as no-op here https://github.com/llvm/llvm-project/commit/565470ed27131b2550e71cf334bf3f33f9d82c62. Not sure we could use it to mark the block live.